testsuite/features/step_definitions/common_steps.rb

```diff
@@ -936,3 +936,29 @@ def server_secret
   count_failures = failings.length
   raise "\nFailures in event history found:\n\n#{failings}" if count_failures.nonzero?
 end
+
+And(/^I wait until all events in history are completed$/) do
+  steps %(
+    When I follow "Events" in the content area
+    And I follow "History" in the content area
+    Then I should see a "System History" text
+  )
+  begin
+    Timeout.timeout(DEFAULT_TIMEOUT) do
+      loop do
+        pickedup = ""
+        events_icons = "//div[@class='table-responsive']/table/tbody/tr/td[2]/i"
+        events = all(:xpath, events_icons)
+        events.each do |ev|
+          if ev[:class].include?('fa-exchange')
+            pickedup << "#{tr.text}\n"
```

Comment:
if you don't want to use the values of `pickedup` strings, there's also the solution to use a boolean:

`found = true`

and even abort prematurely the inner loop.

But maybe it's nice to have the picked up texts for debugging purposes, so feel free to ignore.